### PR TITLE
SDIT-1432: ✨ Target java 21 and set use web token default to true

### PR DESCRIPTION
**Pull request by petergphillips**
- SDIT-1432: ✨ Target java 21 and set use web token default to true
- SDIT-1432: ✨ Add in no signing for local publishing

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 plugins {
   id("io.github.gradle-nexus.publish-plugin") version "1.1.0"
+  kotlin("jvm") version "1.9.22" apply false
```
Comment (petergphillips) [inline build.gradle.kts:3]: stop warnings in build about kotlin subprojects

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt [hunk 2]
```diff
@@ -9,7 +9,7 @@ data class HmppsSqsProperties(
   val localstackUrl: String = "http://localhost:4566",
   val queues: Map<String, QueueConfig> = mapOf(),
   val topics: Map<String, TopicConfig> = mapOf(),
-  val useWebToken: Boolean = false,
+  val useWebToken: Boolean = true,
```
Comment (petergphillips) [inline hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt:12]: change default

File: test-app-reactive/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagenameasync/config/ResourceServerConfiguration.kt [hunk 3]
```diff
@@ -13,19 +14,18 @@ import org.springframework.security.web.server.SecurityWebFilterChain
...
-          "/webjars/**", "/favicon.ico", "/csrf",
-          "/health/**", "/info", "/h2-console/**",
-          "/v3/api-docs/**", "/swagger-ui/**", "/swagger-ui.html",
-          "/queue-admin/retry-all-dlqs",
-        ).permitAll()
-          .anyExchange().authenticated()
-      }
-      .oauth2ResourceServer { it.jwt().jwtAuthenticationConverter(AuthAwareTokenConverter()) }
-      .build()
+  fun springSecurityFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain = http {
```
Comment (petergphillips) [inline test-app-reactive/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagenameasync/config/ResourceServerConfiguration.kt:17]: fix deprecated warnings

File: test-app/build.gradle.kts [hunk 4]
```diff
@@ -9,7 +9,7 @@ configurations {
 
 dependencies {
   annotationProcessor("org.springframework.boot:spring-boot-configuration-processor")
-  implementation("org.springframework.boot:spring-boot-starter-webflux")
+  implementation("org.springframework.boot:spring-boot-starter-web")
```
Comment (petergphillips) [inline test-app/build.gradle.kts:12]: not sure why this wasn't `web` in the first place

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**